.github/workflows/bundle-size.yml

```diff
@@ -0,0 +1,113 @@
+  AZURE_SUBSCRIPTION: Azure PR deploy - NEW
+  AZURE_STORAGE: fluentuipr
+
+jobs:
+  bundle-size:
+    runs-on: macos-14-xlarge
+    permissions:
+      contents: 'read'
+      actions: 'read'
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v4
+        with:
+          main-branch-name: 'master'
+
+      - uses: actions/setup-node@v4
+        with:
+          cache: 'yarn'
+          node-version: '20'
+
+      - run: echo number of CPUs "$(getconf _NPROCESSORS_ONLN)"
+
+      - run: yarn install --frozen-lockfile
+
+      - name: Build packages & create reports (PR only)
+        if: github.event_name == 'pull_request'
+        run: yarn nx affected -t bundle-size --nxBail
+
+      - name: Compare bundle size with base (PR only)
+        if: github.event_name == 'pull_request'
+        run: npx monosize compare-reports --branch=${{ github.event.pull_request.base.ref }} --output=markdown --quiet > ./monosize-report.md
+
+      - name: Save PR number
+        run: echo ${{ github.event.number }} > pr.txt
+
+      - uses: actions/upload-artifact@v4
+        if: ${{ github.event_name == 'pull_request' }}
+        with:
+          name: monosize-report
+          if-no-files-found: error
+          path: |
+            monosize-report.md
+            pr.txt
+
+      # ===================================
+      #   NON PR (MASTER) WORKFLOW STEPS
+      # ===================================
+
+      - name: Build all packages & create reports (non-PR)
+        if: github.event_name != 'pull_request'
+        run: yarn nx run-many -t bundle-size --nxBail
+
+      - name: Login via Azure CLI (non-PR)
+        if: github.event_name != 'pull_request'
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Upload a report (non-PR)
+        if: github.event_name != 'pull_request'
+        uses: azure/cli@v2
+        with:
+          azcliversion: latest
+          # TODO: how to set subcription? monosize has no flag for it and gha az cli doesn't support this parameter
+          # azureSubscription: 'Bundlesize Testing'
+          inlineScript: |
+            yarn monosize upload-report --branch=${{ github.ref }} --commit-sha ${{ github.sha }}
+        env:
+          AZCOPY_AUTO_LOGIN_TYPE: 'AZCLI'
+          SYSTEM_ACCESSTOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # AZURE_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+          # AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+          # AZURE_SERVICE_CONNECTION_ID: ${{ secrets.AZURE_SERVICE_CONNECTION_ID }}
+          # BUNDLESIZE_ACCOUNT_NAME: ${{ secrets.BUNDLESIZE_ACCOUNT_NAME }}
```

Comment:
This needs to be filled in with the storage account name to which the Azure App has access to.